sa/model.go

```diff
@@ -0,0 +1,56 @@
+package sa
+
+import (
+	"encoding/json"
+	"fmt"
+
+	jose "github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/letsencrypt/go-jose"
+	"github.com/letsencrypt/boulder/core"
+)
+
+// regModel is the description of a core.Registration in the database.
+type regModel struct {
+	ID        int64           `db:"id"`
+	Key       []byte          `db:"jwk"`
+	KeySHA256 string          `db:"jwk_sha256"`
+	Contact   []*core.AcmeURL `db:"contact"`
+	Agreement string          `db:"agreement"`
+	LockCol   int64
+}
+
+// newReg creates a reg model object from a core.Registration
+func registrationToModel(r *core.Registration) (*regModel, error) {
+	key, err := json.Marshal(r.Key)
+	if err != nil {
+		return nil, err
+	}
+
+	sha, err := core.KeyDigest(r.Key) // FIXME KeyDigest to []byte, fix lookup in GetRegistrationByKey
```

Comment:
I did intend to fix it (probably by just deleting the FIXME comment). My regrets. Will fix.